core/domain/question_services.py

```diff
@@ -407,14 +406,15 @@ def replace_skill_id_for_all_questions(
     question_models.QuestionSkillLinkModel.put_multi_question_skill_links(
         new_question_skill_link_models)
 
-    questions = get_questions_by_ids(list(question_ids))
-    for question in questions:
-        new_linked_skill_ids = copy.deepcopy(question.linked_skill_ids)
-        new_linked_skill_ids.remove(curr_skill_id)
-        new_linked_skill_ids.append(new_skill_id)
-        _update_linked_skill_ids_of_question(
-            user_id, question.id, new_linked_skill_ids,
-            question.linked_skill_ids)
+    old_questions = question_models.QuestionModel.get_multi(list(question_ids))
+    new_questions = []
+    for question in old_questions:
+        new_question = copy.deepcopy(question)
+        new_question.linked_skill_ids.remove(curr_skill_id)
+        new_question.linked_skill_ids.append(new_skill_id)
+        new_questions.append(new_question)
+    question_models.QuestionModel.delete_multi_questions(old_questions)
+    question_models.QuestionModel.put_multi_questions(new_questions)
```

Comment:
I think just put_multi() will overwrite, right? So you don't need the delete_multi in this case because the question ID is not changing?

Reply:
Done